rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -337,6 +337,32 @@ object ProcNormalizeMatcher {
         ProcVisitOutputs(input.par.prepend(foldedNew), bodyResult.knownFree)
       }
 
+      case p: PMatch => {
+        import scala.collection.JavaConverters._
+
+        val value = normalizeMatch(p.proc_, input)
+        val cases = p.listcase_.asScala.toList.map {
+          case ci: CaseImpl => (ci.proc_1, ci.proc_2)
+          case _            => throw new Error("Unexpected Case implementation.")
+        }
+
+        val initAcc = (List[Tuple2[Par, Par]](), input.knownFree)
+        val casesResult = (initAcc /: cases) { (acc, caseImpl) =>
+          caseImpl match {
+            case (pattern, caseBody) => {
+              val patternResult =
+                normalizeMatch(pattern, ProcVisitInputs(Par(), input.env, input.knownFree))
```

Comment:
The pattern should be normalized with an empty free map.

Reply:
Done